docs/dev/test-architecture.adoc

```diff
@@ -184,7 +184,13 @@ Context("when running help for storage command", func() {
 ----
 +
 
-* Due to parallel test run support make sure that the should run in isolation, otherwise the test result will lead to race condition. To achieve this ginkgo provides some in build functions `BeforeEach`, `AfterEach` etc.
+* Don't create a new test spec for the steps which can be run with the existing specs. Also try to avoid adding the functionality which are part of another feature test file, only add if it is a key requirement.
```

Comment:
Sentences are a bit weird..

This may be better:

**Do not create a new [Spec](https://onsi.github.io/ginkgo/#structuring-your-specs) if there is already an existing one.**

Instead of what you current have. Both sentences say the same thing (you're just repeating the first sentence in the second sentence).